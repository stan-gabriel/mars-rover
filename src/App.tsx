import React, {useState} from 'react';
import './App.css';
import Rover from "./@types/rover";
import {multipleCommands} from "./utils/commands";

function App() {
    const [rover, setRover] = useState<Rover>({x: 0, y: 0, direction: "EAST"})
    const [commands, setCommands] = useState('')
    
    const handleChange = (event: React.FormEvent<HTMLInputElement>) => {
        const result = event.currentTarget.value.replace(/[^fblr]/gi, '');
        setCommands(result);
    }

    const sendCommands = () => {
        console.log(commands, "- commands");
        const newRoverState = multipleCommands(rover, commands);
      setRover({...newRoverState});
      setCommands('');
    }

    const handleResetRover = () => {
        setRover({x: 0, y: 0, direction: "EAST"});
    }

  return (
    <div className="App">
      <header className="App-header">
          <div>
              <p>{`Rover Coordinates: (${rover.x}, ${rover.y}) ${rover.direction}`}</p>
              <input type='text' value={commands} onChange={handleChange}/>
              <button onClick={sendCommands}>Send commands</button>

              <div style={{fontSize: '0.8rem'}}>
                  <i>Only use: 'F' 'B' 'L' 'R'</i>
              </div>

              <button onClick={handleResetRover}>reset Rover position</button>
          </div>
      </header>
    </div>
  );
}

export default App;
