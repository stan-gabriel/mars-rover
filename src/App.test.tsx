import React from 'react';
import renderer from 'react-test-renderer';

import { render, screen } from '@testing-library/react';
import App from './App';

it('renders Send Commands button', () => {
  render(<App />);
  const sendCommandsBtn = screen.getByText(/Send commands/i);
  expect(sendCommandsBtn).toBeInTheDocument();
});

it('renders Send Reset Rover position', () => {
  render(<App />);
  const resetBtn = screen.getByText(/reset Rover position/i);
  expect(resetBtn).toBeInTheDocument();
});

it('changes the class when hovered', () => {
  const component = renderer.create(
      <App />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

