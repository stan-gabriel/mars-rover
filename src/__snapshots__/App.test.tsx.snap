// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div>
      <p>
        Rover Coordinates: (0, 0) EAST
      </p>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        onClick={[Function]}
      >
        Send commands
      </button>
      <div
        style={
          Object {
            "fontSize": "0.8rem",
          }
        }
      >
        <i>
          Only use: 'F' 'B' 'L' 'R'
        </i>
      </div>
      <button
        onClick={[Function]}
      >
        reset Rover position
      </button>
    </div>
  </header>
</div>
`;
